cli/bench/types.go

```diff
@@ -8,3 +13,44 @@ type Results struct {
 	duration             float64 // Benchmark duration.
 	requestsPerSecond    int     // Cumber of requests per second - the main measured value.
 }
+
+// Interface for various types of requests
+type Request interface {
```

Comment:
Obviously, with such a set of queries, it looks easier to implement through a single structure. However, the requests themselves are still logically different.
In addition, polymorphism has more advantages than it seems at first glance:
- When adding data for one type of query, you do not need to add them when creating other types of queries
- Simple addition of a new type of query without reading into the data fields in other types of queries
- The possibility of simple implementation of combined queries

Reply:
We can make a standard pattern of 50 inserts/50 deletions and do the deletion without searching for the key, since we already know it

Reply:
> We can make a standard pattern of 50 inserts/50 deletes by combining two operations into one and do the deletion without searching for the key, since we already know it

Did I understand you correctly that it is something like `type InsertDelete50Request`?
If so, it is not necessary the load pattern is determined at another level.
I can also note that [here](https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/net/http/request.go;l=110), to determine the type (method) of the request, just a separate field used.